test/Header.test.js

```diff
@@ -31,226 +33,449 @@ const mockPatronApiEndpoint = `${appConfig.patronApiUrl}eyJ0eXAiOiJKV1QiLCJhbGci
+            console.warn(response.headers);
+            console.warn(response.config);
+
+            // If the cookie for getting log in Data is expired
+            if (response.data.statusCode === 401 && response.data.expired === true) {
+              utils.refreshAccessToken(
+                refreshApi,
+                () => {
+                  const patronNameObject = utils.modelPatronName(
+                    utils.extractPatronName(mockResponseData)
+                  );
+
+                  component.setState({
+                    patronName: patronNameObject.name,
+                    patronInitial: patronNameObject.initial,
+                    patronDataReceived: true,
+                  });
+                },
+                refreshFailCb,
+              );
+            }
           }
+        });
+    };
+
+    let mock = null;
+
+    before(() => {
+      mock = new MockAdapter(axios);
+    });
+
+    after(() => {
+      mock.restore();
+    });
+
+    describe('when "nyplIdentityPatron" cookie exists', () => {
+      let component;
+
+      // functions in Header.js
+      let setLoginCookie;
+      let fetchPatronData;
+
+      // functions in utils.js
+      let hasNyplIdentityPatronCookie;
+      let getNyplIdentityPatronCookie;
+      let getPatronData;
+      let modelPatronName;
+
+      before(() => {
+        // functions in Header.js
+        setLoginCookie = sinon.spy(Header.prototype, 'setLoginCookie');
+        fetchPatronData = sinon.spy(Header.prototype, 'fetchPatronData');
+
+        // functions in utils.js
+        hasNyplIdentityPatronCookie = sinon.stub(utils, 'hasCookie')
+          .withArgs('nyplIdentityPatron')
+          .returns(true);
+        getNyplIdentityPatronCookie = sinon.stub(utils, 'getCookie')
+          .withArgs('nyplIdentityPatron')
+          .returns(mockLoginCookie);
+        getPatronData = sinon.spy(utils, 'getLoginData');
+        modelPatronName = sinon.spy(utils, 'modelPatronName');
+
+        // mock up of the API call to get patron's data
+        mock
+          .onGet(mockPatronApiEndpoint)
+          .reply(200, mockResponseData);
+
+        component = mount(<Header />);
+      });
+
+      after(() => {
+        mock.reset();
+        // stubs don't have restore(), the way to restore them is go back to the original functions.
+        // However, if the sutbs only use the methods that belong to spies, restore() will work.
+        utils.hasCookie.restore();
+        utils.getCookie.restore();
+        setLoginCookie.restore();
+        fetchPatronData.restore();
+        getPatronData.restore();
+        modelPatronName.restore();
+      });
+
+      it('should call the function to check if the cookie "nyplIdentityPatron" exists', () => {
+        expect(setLoginCookie.calledOnce).to.equal(true);
+        expect(hasNyplIdentityPatronCookie.calledOnce).to.equal(true);
+        hasNyplIdentityPatronCookie.alwaysCalledWithExactly('nyplIdentityPatron');
```

Comment:
This is a statement and not a test. It returns `true` but even if it returned `false` the test would pass. What you need to do is `expect(hasNyplIdentityPatronCookie.alwaysCalledWithExactly('nyplIdentityPatron')).to.equal(true);`, otherwise you can do something like `hasNyplIdentityPatronCookie.alwaysCalledWithExactly('some fake string');` and the test will still pass.

Reply:
ohh This PR does not focus on this section rather than just lints the syntax. Let me see how complicate to fix it. If too much codes need to changed I will do it in another PR.

Reply:
Actually I will update it.

Reply:
Thanks.